pinot-core/src/test/java/org/apache/pinot/queries/FSTBasedRegexpLikeQueriesTest.java

```diff
@@ -0,0 +1,249 @@
+        fieldConfigs.add(new FieldConfig(
+                DOMAIN_NAMES_COL,
+                FieldConfig.EncodingType.DICTIONARY,
+                FieldConfig.IndexType.FST, null));
+        fieldConfigs.add(new FieldConfig(
+                URL_COL,
+                FieldConfig.EncodingType.DICTIONARY,
+                FieldConfig.IndexType.FST, null));
+
+        TableConfig tableConfig = new TableConfigBuilder(TableType.OFFLINE).setTableName(TABLE_NAME)
+                .setInvertedIndexColumns(Arrays.asList(DOMAIN_NAMES_COL))
+                .setFieldConfigList(fieldConfigs).build();
+        Schema schema = new Schema.SchemaBuilder().setSchemaName(TABLE_NAME)
+                .addSingleValueDimension(DOMAIN_NAMES_COL, FieldSpec.DataType.STRING)
+                .addSingleValueDimension(URL_COL, FieldSpec.DataType.STRING)
+                .addMetric(INT_COL_NAME, FieldSpec.DataType.INT).build();
+        SegmentGeneratorConfig config = new SegmentGeneratorConfig(tableConfig, schema);
+        config.setOutDir(INDEX_DIR.getPath());
+        config.setTableName(TABLE_NAME);
+        config.setSegmentName(SEGMENT_NAME);
+
+        SegmentIndexCreationDriverImpl driver = new SegmentIndexCreationDriverImpl();
+        try (RecordReader recordReader = new GenericRowRecordReader(rows)) {
+            driver.init(config, recordReader);
+            driver.build();
+        }
+    }
+
+    private void testSelectionResults(String query, int expectedResultSize)
+            throws Exception {
+        Operator<IntermediateResultsBlock> operator = getOperatorForPqlQuery(query);
+        IntermediateResultsBlock operatorResult = operator.nextBlock();
+        List<Object[]> resultset = (List<Object[]>) operatorResult.getSelectionResult();
+        Assert.assertNotNull(resultset);
+        Assert.assertEquals(resultset.size(), expectedResultSize);
+    }
+
+    private AggregationGroupByResult getGroupByResults(String query)
+            throws Exception {
+        AggregationGroupByOperator operator = getOperatorForPqlQuery(query);
+        IntermediateResultsBlock resultsBlock = operator.nextBlock();
+        return resultsBlock.getAggregationGroupByResult();
+    }
+
+    private void matchGroupResult(AggregationGroupByResult result, String key, long count) {
+        Iterator<GroupKeyGenerator.GroupKey> groupKeyIterator = result.getGroupKeyIterator();
+        while (groupKeyIterator.hasNext()) {
+            GroupKeyGenerator.GroupKey groupKey = groupKeyIterator.next();
+            Assert.assertEquals(((Number)result.getResultForKey(groupKey, 0)).longValue(), count);
+        }
+    }
+
+    @Test
+    public void testFSTBasedRegexpLike() throws Exception {
+        // Select queries on col with FST + inverted index.
+        String query =
+                "SELECT INT_COL, DOMAIN_NAMES FROM MyTable WHERE REGEXP_LIKE(DOMAIN_NAMES, 'www.domain1.*') LIMIT 50000";
+        testSelectionResults(query, 256);
+
```

Comment:
- You should also add tests for queries where REGEXP_LIKE filter is combined with another filter(s) using AND/OR to ensure the docID intersection/union is working fine. 
- Also combine multiple REGEXP_LIKE using a mix of FST index and no FST index. 
- See tests in TextSearchQueriesTest for examples

Reply:
done, ptal